spec/integration/aws_rds_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
+require 'cheffish/rspec/chef_run_support'
+require 'chef/provisioning/aws_driver/credentials'
+
+describe 'Aws RDS DB instance' do
+  extend Cheffish::RSpec::ChefRunSupport
+
+  when_the_chef_12_server 'exists' do
+    organization 'foo'
+
+    let(:rds_client) { double(AWS::RDS::Client) }
+    let!(:entry_store) { Chef::Provisioning::ChefManagedEntryStore.new }
+
+    before :each do
+      Chef::Config.chef_server_url = URI.join(Chef::Config.chef_server_url, '/organizations/foo').to_s
+      allow_any_instance_of(AWS.config.class).to receive(:rds_client).and_return(rds_client)
+      allow(Chef::Provisioning::ChefManagedEntryStore).to receive(:new).and_return(entry_store)
+      allow_any_instance_of(Chef::Provisioning::AWSDriver::Credentials).to receive(:default)
+        .and_return({
+          :aws_access_key_id => 'na',
+          :aws_secret_access_key => 'na'
+        })
+    end
+
+    describe 'action :create' do
+
+      before do
+        resp = AWS::Core::Response.new
+
+        expect(rds_client).to receive(:describe_db_instances)
+          .with({:db_instance_identifier=>'my-db-instance'})
+          .and_raise(AWS::RDS::Errors::DBInstanceNotFound)
+
```

Comment:
I thought it was weird as well, I think I copied the style from the vpc spec. Not a hardcore rubyist (yet), so I am not sure what the ruby way is ;-)

Op 21 mrt. 2015 16:00 schreef Drew Blessing <notifications@github.com>:

In spec/integration/aws_rds_spec.rb<https://github.com/chef/chef-provisioning-aws/pull/119#discussion_r26892562>:

> +      allow_any_instance_of(Chef::Provisioning::AWSDriver::Credentials).to receive(:default)
> +        .and_return({
> +          :aws_access_key_id => 'na',
> +          :aws_secret_access_key => 'na'
> +        })
> +    end
> +
> +    describe 'action :create' do
> +
> +      before do
> +        resp = AWS::Core::Response.new
> +
> +        expect(rds_client).to receive(:describe_db_instances)
> +          .with({:db_instance_identifier=>'my-db-instance'})
> +          .and_raise(AWS::RDS::Errors::DBInstanceNotFound)
> +


Is it a good idea to put expectations in before/after blocks? I've never seen that done. Seems like they ought to be in 'it' blocks?

—
Reply to this email directly or view it on GitHub<https://github.com/chef/chef-provisioning-aws/pull/119/files#r26892562>.

[https://www.evision-software.com/wp-content/uploads/2015/03/bannersmall2.png]